### PR TITLE
Move DiagnosticsHandler.IsGloballyEnabled back to a method

### PR DESCRIPTION
**Pull request by steveisok**
Intent is to make linker substitutions happy and try to unblock https://github.com/dotnet/sdk/pull/18801

### CODE REVIEW

**Comment by steveisok** [comment]
@vitek-karas @eerhardt Given the current code structure, is the linker substitution going to work?

**Review by jeffschwMSFT — APPROVED** [review]
lgtm.  Thanks @steveisok

**Review by vitek-karas — APPROVED** [review]
You could change it to a property, you just need to update the ILLink.Substitutions.xml for that assembly to now point to the property getter - that should work as well.

**Review by eerhardt — APPROVED** [review]

File: src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs [hunk 1]
```diff
@@ -31,10 +31,13 @@ internal static bool IsEnabled()
...
             // or if someone listens to HttpHandlerDiagnosticListener
-            return IsGloballyEnabled && (Activity.Current != null || s_diagnosticListener.IsEnabled());
+            return IsGloballyEnabled() && (Activity.Current != null || s_diagnosticListener.IsEnabled());
         }
 
-        internal static bool IsGloballyEnabled => GlobalHttpSettings.DiagnosticsHandler.EnableActivityPropagation;
+        internal static bool IsGloballyEnabled()
+        {
+            return GlobalHttpSettings.DiagnosticsHandler.EnableActivityPropagation;
+        }
```
Comment (ManickaP) [inline src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs:40]:
```suggestion
        internal static bool IsGloballyEnabled() => GlobalHttpSettings.DiagnosticsHandler.EnableActivityPropagation;
```

**Review by ManickaP — APPROVED** [review]
LGTM
cc @MihaZupan

**Review by MihaZupan — APPROVED** [review]

**Pull request merged**